src/pages/index.js

```diff
@@ -0,0 +1,296 @@
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  // set card details
+  cardNameEl.textContent = data.name;
+  console.log(data);
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Handle like button click
+  cardLikedBtn.addEventListener("click", (evt) =>
+    handleLike(evt, data._id, data.isLiked)
+  );
+  if (data.isLiked) {
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  // Handle delete button click
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  // Handle image preview modals
+  cardImageEl.addEventListener("click", () => {
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+function handleClickOverlay(evt) {
+  if (evt.target.classList.contains("modal_opened")) {
+    closeModal(evt.target);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keyup", handleEscClose);
+  document.addEventListener("click", handleClickOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", handleEscClose);
+  document.removeEventListener("click", handleClickOverlay);
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    closeModal(activeModal);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const editProfilesubmitButton = evt.submitter;
+  setButtonText(editProfileSubmitButton, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDesriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setTimeout(() => {
+        setButtonText(editProfileSubmitButton, false);
+      }, 300);
+    });
+}
+
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const cardSaveBtn = evt.target.querySelector(".modal__submit-btn");
+  setButtonText(cardSaveBtn, true);
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardslist.prepend(cardElement);
+      closeModal(cardModal);
+      disabledButton(cardSaveBtn, validationConfig);
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setTimeout(() => {
+        setButtonText(cardSaveBtn, false);
+      }, 300);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSaveBtn = evt.target.querySelector(".modal__submit-btn");
```

Comment:
`const avatarSaveBtn  = evt.submitter;` can be used inside any submit handler as the save button. You don't need to search it 


```js
  const avatarSaveBtn = evt.submitter

```